File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    picks =[
    [1,2,3,4,5],
    [2, 1, 2, 3, 2, 4, 2, 5],
    [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]]
    corrects = []
    for pick in picks:
        correct = 0
        for idx in range(len(answers)):
            if (pick[idx%len(pick)]==answers[idx]):
                ## print(pick, idx%len(pick))
                correct+=1
        corrects.append(correct)
    max_v = max(corrects)
    answer = []
    for idx in range(len(corrects)):
        if (corrects[idx]==max_v):
            answer.append(idx+1)
            
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    const picks = [[1, 2, 3, 4, 5],
                   [2, 1, 2, 3, 2, 4, 2, 5],
                   [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
                 ]
    const P = picks.length
    let scores = [0,0,0]
    for(let idx = 0;idx<P;idx++){
        const pick = picks[idx]
        for(let i=0;i<answers.length;i++){
            if(pick[i%pick.length] == answers[i]){
                
                scores[idx]+=1
            }
            
        }
    }
    const max_score = Math.max(...scores)
   
    const answer = []
    for (let idx = 0;idx<scores.length;idx++){
        if(scores[idx]==max_score){
            answer.push(idx+1)
        }
        
    }
    return answer;
}